### PR TITLE
[SITE] Add preview images for blocks 

### PR DESCRIPTION
**Pull request by teenoh**
This PR 
- updates the codebase to pull block preview images from the block's metadata file
- adds some styling improvements to the block card and block slider components.

How to test this
- Change the `branch` property in the registry files (`hub/*.json`)  to `vu/block-preview-images`
- Run `yarn build-blocks` from the `site/` directory (might need to delete the `public/blocks` folder before doing this)
- Run `yarn dev`
- The preview images should show up when you go to the hub page ("/hub")

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201555682469130

### CODE REVIEW

**Comment by teenoh** [comment]
Needs to be merged in after https://github.com/hashintel/dev/pull/476

File: site/src/pages/api/blocks.api.ts [hunk 1]
```diff
@@ -107,11 +66,10 @@ export const readBlocksFromDisk = (): BlockMetadata[] => {
     }))
     .map((metadata: BlockMetadata) => ({
       ...metadata,
+      image: `/blocks/${metadata.packagePath}/${metadata.image}`,
```
Comment (CiaranMn) [inline site/src/pages/api/blocks.api.ts:69]: This assumes that the image URL is relative - can we also handle absolute URLs please?

e.g. check here if it starts with `/`, and if so do this, and if not just include the URL?

Can we also make the same fix for icons?

**Review by CiaranMn — APPROVED** [review]
Thanks @teenoh - small fix requested, good to merge with that in

File: site/src/pages/api/blocks.api.ts (hunk 1 above)
Comment (teenoh) [reply site/src/pages/api/blocks.api.ts]: Added in this commit 1146653

**Pull request merged**